src/Lecture1.hs

```diff
@@ -108,7 +123,11 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Integer
+strSum str = sum ints
+    where ints = map toInt strings
+          toInt s = read s :: Integer
+          strings = words str
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body.

As a consequence, you can write this function as a pretty one-liner 🙃 

```suggestion
strSum str = sum (map read (words str))
```